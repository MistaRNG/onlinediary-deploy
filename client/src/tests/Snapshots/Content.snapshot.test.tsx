import React from 'react';
import { render } from '@testing-library/react';
import Content from '../../features/journal/components/Main';
import '@testing-library/jest-dom';
import { describe, it, expect, jest } from '@jest/globals';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';

jest.mock('../../features/journal/components/Editor', () => () => <div>Editor Mock</div>);

const mockReducer = () => ({
  journals: {
    data: {},
  },
});

const mockStore = configureStore({
  reducer: {
    journals: mockReducer,
  },
});

describe('Content Component Snapshot Tests', () => {
  it('should render the Content component correctly', () => {
    // ARRANGE
    const mockDisabledDays = () => false;
    const mockMinDate = null;
    const mockValue = null;
    const mockDate = '';
    const mockDates = [];

    // ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <BrowserRouter>
          <Content
            disabledDays={mockDisabledDays}
            minDate={mockMinDate}
            value={mockValue}
            date={mockDate}
            dates={mockDates}
          />
        </BrowserRouter>
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
