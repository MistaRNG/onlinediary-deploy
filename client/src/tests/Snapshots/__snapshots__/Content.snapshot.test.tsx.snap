// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component Snapshot Tests should render the Content component correctly 1`] = `
<DocumentFragment>
  <div
    class="Journal"
  >
    <div>
      <div
        class="search"
      >
        <div
          class="bp4-input-group bp4-large search-input"
        >
          <input
            class="bp4-input"
            placeholder="Search..."
            style="padding-right: 0px;"
            type="text"
            value=""
          />
          <span
            class="bp4-input-action"
          />
        </div>
        <div
          class="search-result"
        >
          No journal is found
        </div>
      </div>
      <div
        class="calendar-container"
      >
        <div
          class="calendar-header"
        >
          <button
            class="today-button"
          >
            Today
          </button>
        </div>
        <div
          style="display: contents;"
        >
          <div
            aria-label="Choose Date"
            aria-modal="true"
            class="react-datepicker custom-calendar"
            role="dialog"
          >
            <span
              aria-live="polite"
              class="react-datepicker__aria-live"
              role="alert"
            />
            <button
              aria-label="Previous Month"
              class="react-datepicker__navigation react-datepicker__navigation--previous"
              type="button"
            >
              <span
                class="react-datepicker__navigation-icon react-datepicker__navigation-icon--previous"
              >
                Previous Month
              </span>
            </button>
            <button
              aria-label="Next Month"
              class="react-datepicker__navigation react-datepicker__navigation--next"
              type="button"
            >
              <span
                class="react-datepicker__navigation-icon react-datepicker__navigation-icon--next"
              >
                Next Month
              </span>
            </button>
            <div
              class="react-datepicker__month-container"
            >
              <div
                class="react-datepicker__header "
              >
                <h2
                  class="react-datepicker__current-month"
                >
                  September 2024
                </h2>
                <div
                  class="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
                />
                <div
                  class="react-datepicker__day-names"
                >
                  <div
                    aria-label="Sunday"
                    class="react-datepicker__day-name"
                  >
                    Su
                  </div>
                  <div
                    aria-label="Monday"
                    class="react-datepicker__day-name"
                  >
                    Mo
                  </div>
                  <div
                    aria-label="Tuesday"
                    class="react-datepicker__day-name"
                  >
                    Tu
                  </div>
                  <div
                    aria-label="Wednesday"
                    class="react-datepicker__day-name"
                  >
                    We
                  </div>
                  <div
                    aria-label="Thursday"
                    class="react-datepicker__day-name"
                  >
                    Th
                  </div>
                  <div
                    aria-label="Friday"
                    class="react-datepicker__day-name"
                  >
                    Fr
                  </div>
                  <div
                    aria-label="Saturday"
                    class="react-datepicker__day-name"
                  >
                    Sa
                  </div>
                </div>
              </div>
              <div
                aria-label="Month September, 2024"
                class="react-datepicker__month"
                role="listbox"
              >
                <div
                  class="react-datepicker__week"
                >
                  <div
                    aria-disabled="false"
                    aria-label="Choose Sunday, September 1st, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--001 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    1
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Monday, September 2nd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--002"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    2
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Tuesday, September 3rd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--003"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    3
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Wednesday, September 4th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--004"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    4
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Thursday, September 5th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--005"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    5
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Friday, September 6th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--006"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    6
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Saturday, September 7th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--007 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    7
                  </div>
                </div>
                <div
                  class="react-datepicker__week react-datepicker__week--keyboard-selected"
                >
                  <div
                    aria-current="date"
                    aria-disabled="false"
                    aria-label="Choose Sunday, September 8th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day  react-datepicker__day--today react-datepicker__day--008 react-datepicker__day--keyboard-selected react-datepicker__day--today react-datepicker__day--weekend"
                    role="option"
                    tabindex="0"
                    title=""
                  >
                    8
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Monday, September 9th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--009"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    9
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Tuesday, September 10th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--010"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    10
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Wednesday, September 11th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--011"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    11
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Thursday, September 12th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--012"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    12
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Friday, September 13th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--013"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    13
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Saturday, September 14th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--014 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    14
                  </div>
                </div>
                <div
                  class="react-datepicker__week"
                >
                  <div
                    aria-disabled="false"
                    aria-label="Choose Sunday, September 15th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--015 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    15
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Monday, September 16th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--016"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    16
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Tuesday, September 17th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--017"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    17
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Wednesday, September 18th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--018"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    18
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Thursday, September 19th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--019"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    19
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Friday, September 20th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--020"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    20
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Saturday, September 21st, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--021 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    21
                  </div>
                </div>
                <div
                  class="react-datepicker__week"
                >
                  <div
                    aria-disabled="false"
                    aria-label="Choose Sunday, September 22nd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--022 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    22
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Monday, September 23rd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--023"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    23
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Tuesday, September 24th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--024"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    24
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Wednesday, September 25th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--025"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    25
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Thursday, September 26th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--026"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    26
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Friday, September 27th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--027"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    27
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Saturday, September 28th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--028 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    28
                  </div>
                </div>
                <div
                  class="react-datepicker__week"
                >
                  <div
                    aria-disabled="false"
                    aria-label="Choose Sunday, September 29th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--029 react-datepicker__day--weekend"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    29
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Monday, September 30th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--030"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    30
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Tuesday, October 1st, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    1
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Wednesday, October 2nd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    2
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Thursday, October 3rd, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--003 react-datepicker__day--outside-month"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    3
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Friday, October 4th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--004 react-datepicker__day--outside-month"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    4
                  </div>
                  <div
                    aria-disabled="false"
                    aria-label="Choose Saturday, October 5th, 2024"
                    aria-selected="false"
                    class="react-datepicker__day react-datepicker__day--005 react-datepicker__day--weekend react-datepicker__day--outside-month"
                    role="option"
                    tabindex="-1"
                    title=""
                  >
                    5
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      Editor Mock
    </div>
  </div>
</DocumentFragment>
`;
